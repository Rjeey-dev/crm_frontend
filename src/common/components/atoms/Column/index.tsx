import React, {ReactNode, ReactNodeArray} from "react";

interface IProps {
    classes?: string,
    children: ReactNodeArray | ReactNode,
}

function Column(props: IProps) {
    return <div className={props.classes}>
        {props.children}
    </div>;
}

Column.defaultProps = {
    classes: 'col-sm-9'
};

export default Column;