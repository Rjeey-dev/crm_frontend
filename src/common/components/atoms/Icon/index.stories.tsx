import React from 'react';
import { storiesOf } from '@storybook/react';
import Icon from './index';

storiesOf('Icon', module)
    .add('notifications', () => (
        <Icon classes='mdi mdi-file-document-box'/>
    ))
    .add('settings', () => (
        <Icon classes='mdi mdi-tune'/>
    ));