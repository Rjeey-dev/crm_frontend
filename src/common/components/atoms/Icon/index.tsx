import React from 'react';

interface IProps {
    classes: string,
    attributes?: object
}

function Icon(props: IProps) {
    return <i className={props.classes} {...props.attributes}/>;
}

export default Icon;
