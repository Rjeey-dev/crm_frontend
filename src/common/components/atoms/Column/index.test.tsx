import React from 'react';
import { shallow } from 'enzyme';

import Icon from '../Icon/index';
import Column from './index';

it("Renders <Column/> with correct properties", () => {
    const output = shallow(
        <Column classes='col-sm-6'>
            <Icon classes='icon'/>
        </Column>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <Column/> with only required props", () => {
    const output = shallow(
        <Column>
            <Icon classes='icon'/>
        </Column>
    ).html();

    expect(output).toMatchSnapshot();
});