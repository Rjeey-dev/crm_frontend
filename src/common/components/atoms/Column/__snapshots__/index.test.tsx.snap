// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Column/> with correct properties 1`] = `"<div class=\\"col-sm-6\\"><i class=\\"icon\\"></i></div>"`;

exports[`Renders <Column/> with only required props 1`] = `"<div class=\\"col-sm-9\\"><i class=\\"icon\\"></i></div>"`;
